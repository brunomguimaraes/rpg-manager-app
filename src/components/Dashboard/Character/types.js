// @flow

export type Character = {
  age: number,
  avatar: string,
  className: string,
  gameId: number,
  healthPoints: number,
  id: number,
  level: number,
  magicPoints: number,
  name: string,
  player: string,
  race: string
};
