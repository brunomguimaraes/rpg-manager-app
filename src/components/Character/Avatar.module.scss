@import "~styles/variables";

.defaultAvatar {
  overflow: hidden;
  position: relative;
  width: $spacer * 15;
  height: $spacer * 15;
  border-right: 1px solid black;
  @media screen and (max-width: $break-small) {
    width: $spacer * 10;
    height: $spacer * 10;
  }
  & > img {
    position: absolute;
    width: $spacer * 15;
    height: $spacer * 15;
    transform: translate(0%, 0%);
    @media screen and (max-width: $break-small) {
      width: $spacer * 10;
      height: $spacer * 10;
    }
  }
}

.avatar {
  overflow: hidden;
  position: relative;
  width: $spacer * 15;
  height: $spacer * 15;
  box-sizing: border-box;
  border-right: 1px solid black;
  @media screen and (max-width: $break-small) {
    width: $spacer * 10;
    height: $spacer * 10;
  }
  & > img {
    position: absolute;
    left: 50%;
    top: 85%;
    height: 100%;
    width: auto;
    transform: translate(-50%, -50%);
    @media screen and (max-width: $break-small) {
      top: 120%;
    }
    &.portrait {
      width: 100%;
      height: auto;
      @media screen and (max-width: $break-small) {
        width: 150%;
        height: auto;
      }
    }
  }
}
