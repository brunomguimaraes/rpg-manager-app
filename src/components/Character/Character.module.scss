@import "~styles/variables";

.character {
  display: flex;
  box-sizing: border-box;
  border: 1px solid black;
  &Status {
    flex: 1;
    padding: $spacer;
    & > p {
      display: flex;
      margin: 0 0 $spacer / 2;
    }
    img {
      width: 18px;
      height: 18px;
      margin-right: $spacer;
    }
    span {
      margin-right: $spacer * 2;
    }
  }
  &Player {
    float: right;
  }
}
