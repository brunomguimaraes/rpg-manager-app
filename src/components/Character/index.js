// @flow

export { default } from "./Character";
export type { Character } from "./types";
