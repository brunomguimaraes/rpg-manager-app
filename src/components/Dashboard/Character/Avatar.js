// @flow

import React from "react";
import defaultAvatar from "~/assets/default-avatar.png";
import styles from "./Avatar.module.scss";

type Props = {
  src: string
};

export default function Avatar({ src }: Props) {
  return src ? (
    <picture className={styles.avatar}>
      <img className={styles.portrait} alt="Character avatar." src={src} />
    </picture>
  ) : (
    <picture className={styles.defaultAvatar}>
      <img
        className={styles.portrait}
        alt="Character avatar."
        src={defaultAvatar}
      />
    </picture>
  );
}
