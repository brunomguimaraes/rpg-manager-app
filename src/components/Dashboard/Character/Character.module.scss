@import "~styles/variables";

.character {
  box-sizing: border-box;
  border: $border;
  box-shadow: $box-shadow;
  display: flex;

  .status {
    flex: 1;
    padding: $spacer;
    background: $color4;
    & > p {
      display: flex;
      margin: 0 0 $spacer / 2;
    }
    img {
      width: 18px;
      height: 18px;
      margin-right: $spacer;
    }
    span {
      margin-right: $spacer * 2;
    }
    p,
    span {
      color: $color2;
    }
  }
}

.player {
  float: right;
}
