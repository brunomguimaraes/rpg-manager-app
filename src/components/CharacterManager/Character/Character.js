// @flow

import React from "react";
import Avatar from "./Avatar";
import { Link } from "react-router-dom";
import type { Character as Props } from "./types";
import styles from "./Character.module.scss";
import screen from "~/styles/screen.module.scss";
import mana from "~/assets/mana2.png";
import health from "~/assets/health2.png";

export default function Character(props: Props) {
  return (
    <Link
      className={styles.card}
      to={`/game/${props.gameId}/character/${props.id}`}
    >
      <section className={styles.character}>
        <Avatar src={props.avatar} />
        <div className={styles.status}>
          <p className={styles.right}>
            <strong>Player</strong>: {props.player}
          </p>
          <p>
            <strong>{props.name}</strong>
          </p>
          <p>
            {props.characterClass}, Lv: {props.level}
          </p>
          <p>
            {props.race}, {props.age} years old
          </p>
          <p className={screen.hideOnSmall}>
            <img alt="Health Points" src={health} />
            <span>{props.healthPoints}</span>
            <img alt="Magic Points" src={mana} />
            <span>{props.magicPoints}</span>
          </p>
        </div>
      </section>
    </Link>
  );
}
