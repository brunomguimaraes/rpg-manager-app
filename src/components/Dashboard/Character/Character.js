// @flow

import React from "react";
import Avatar from "./Avatar";
import type { Character as Props } from "./types";
import styles from "./Character.module.scss";
import screen from "~/styles/screen.module.scss";
import mana from "~/assets/mana2.png";
import health from "~/assets/health2.png";

export default function Character(props: Props) {
  return (
    <section className={styles.character}>
      <Avatar src={props.avatar} />
      <div className={styles.status}>
        <p className={`${screen.hideOnMedium} ${styles.player}`}>
          <strong>Player</strong>: {props.player}
        </p>
        <p>
          <strong>{props.name}</strong>
        </p>
        <p>
          {props.className}, Lv: {props.level}
        </p>
        <p>
          {props.race}, {props.age} years old
        </p>
        <p className={screen.hideOnSmall}>
          <img alt="Health Points" src={health} />
          <span>{props.healthPoints}</span>
          <img alt="Magic Points" src={mana} />
          <span>{props.magicPoints}</span>
        </p>
      </div>
    </section>
  );
}
