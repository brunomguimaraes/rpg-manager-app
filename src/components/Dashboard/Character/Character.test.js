import React from "react";
import Character from "./Character";

const character = {
  age: 10,
  avatar: "http://path-to-avatar",
  className: "Rogue",
  gameId: 1,
  healthPoints: 200,
  id: 1,
  level: 20,
  magicPoints: 50,
  name: "Batman",
  player: "Bruce Waine",
  race: "Human"
};

test("Character", () =>
  expect(shallow(<Character {...character} />)).toMatchSnapshot());
