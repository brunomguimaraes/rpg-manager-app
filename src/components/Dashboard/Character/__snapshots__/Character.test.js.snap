// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character 1`] = `
<section
  className="character"
>
  <Avatar
    src="http://path-to-avatar"
  />
  <div
    className="status"
  >
    <p
      className="hideOnMedium player"
    >
      <strong>
        Player
      </strong>
      : 
      Bruce Waine
    </p>
    <p>
      <strong>
        Batman
      </strong>
    </p>
    <p>
      Rogue
      , Lv: 
      20
    </p>
    <p>
      Human
      , 
      10
       years old
    </p>
    <p
      className="hideOnSmall"
    >
      <img
        alt="Health Points"
        src="health2.png"
      />
      <span>
        200
      </span>
      <img
        alt="Magic Points"
        src="mana2.png"
      />
      <span>
        50
      </span>
    </p>
  </div>
</section>
`;
